scripts/morty/isobar_morty_pde.py

```diff
@@ -0,0 +1,502 @@
+        self.S['b'] = self._format_spatial((self.conc_a*self.lama
+                                            + self.FYb/self.vol1),
+                                           (self.conc_a*self.lama),
+                                           vector_form=vector_form)
+        self.S['c'] = self._format_spatial((self.conc_b*self.lamb
+                                            + self.FYc/self.vol1),
+                                           (self.conc_b*self.lama),
+                                           vector_form=vector_form)
+        self.S['d_m1'] = self._format_spatial((self.FYd_m1/self.vol1),
+                                              (0/self.vol2))
+        self.S['d'] = self._format_spatial((self.br_c_d*self.conc_c*self.lamc
+                                            + self.FYd/self.vol1
+                                            + self.br_dm1_d
+                                            * self.conc_d_m1
+                                            * self.lamd_m1),
+                                           (self.br_c_d*self.conc_c*self.lamc
+                                            + self.br_dm1_d
+                                            * self.conc_d_m1
+                                            * self.lamd_m1),
+                                           vector_form=vector_form)
+        return
+
+    def _update_result_mat(self, result_mat, ti):
+        """
+        Updates the result matrix with new concentrations
+
+        Parameters
+        ----------
+        result_mat : 3D matrix
+            Holds values over time, space, and nuclide (in that order)
+        ti : int
+            Current time index
+
+        Returns
+        -------
+        result_mat : 3D matrix
+            Holds values over time, space, and nuclide (in that order)
+        """
+        result_mat[ti, :, 0] = self.conc_a
+        result_mat[ti, :, 1] = self.conc_b
+        result_mat[ti, :, 2] = self.conc_c
+        result_mat[ti, :, 3] = self.conc_d_m1
+        result_mat[ti, :, 4] = self.conc_d
+        return result_mat
+
+    def serial_MORTY_solve(self):
+        """
+        Run MORTY, calculating the isobar concentrations in-core and ex-core
+            for the given problem.
+
+        Returns
+        -------
+        result_mat : 3D matrix
+            Holds values over time, space, and nuclide (in that order)
+
+        """
+        result_mat = self._initialize_result_mat()
+        for ti, t in enumerate(ts[:-1]):
+
+            self._update_sources()
+
+            self.conc_a = self._external_PDE_no_step(self.conc_a, 'a')
+            self.conc_b = self._external_PDE_no_step(self.conc_b, 'b')
+            self.conc_c = self._external_PDE_no_step(self.conc_c, 'c')
+            self.conc_d_m1 = self._external_PDE_no_step(self.conc_d_m1, 'd_m1')
+            self.conc_d = self._external_PDE_no_step(self.conc_d, 'd')
+
+            result_mat = self._update_result_mat(result_mat, ti)
+
```

Comment:
On god. Your life would be so much easier if you stopped hard-coding everything. This is unmaintainable.

Reply:
please run the following commands in your terminal

```bash
ipython
[1]: import this
```

Thanks.